Graphics lab 2

leins275 opened the pull request:
1. Camera movements
2. Average luminance adaptation
3. Light points
4. Eye adaptation

leins275 committed:
Created light project from triangle

leins275 committed:
Merge branch 'main' of https://github.com/PrimatElite/gpu-labs into graphics-lab-2

leins275 committed:
Merge branch 'main' of https://github.com/PrimatElite/gpu-labs into graphics-lab-2

maik-nack committed:
Fixed output directories

maik-nack committed:
Separated code for files and created some classes

maik-nack committed:
Changed project properties

maik-nack committed:
Added simple texture

ExpressFromSiberia committed:
Added 3 point lights

maik-nack committed:
Added HDR with tone mapping

maik-nack committed:
Changed passing data in some methods

maik-nack committed:
Added viewport to device resources

maik-nack committed:
Renamed shaders

maik-nack committed:
Added new utils

leins275 committed:
Added average luminance calculation

maik-nack committed:
Added sRGB

maik-nack committed:
Changed lights

maik-nack committed:
Added eye adaptation?

ExpressFromSiberia committed:
Added test camera

maik-nack committed:
Changed camera

maik-nack committed:
Changed eye adaptation

maik-nack committed:
Changed lights

anthark commented:
Молодцы, почти получилось. 

1) DirectX debug layer сыпет ошибкой про работу с замапленным ресурсом. Включите break на error от DirectX и сразу увидите где, и что сделано неправильно. 
2) В шейдере ps_luminance_main есть небольшой, но смешной косячок - зачем нужно "+ 1" при вычислении логарифмического значения яркости? И что будет на выходе в render target, если там +1 вне логарифма? 
3) Вы читаете float значение для luminance из текстуры формата R16G16B16A16. Но там же значения хранятся из двух байт, а не из четырех. И в итоге в luminance получается как будто мусор. 
4) Забыли, что diffuse текстура должна читаться как sRGB. У вас даже в функции из библиотеки для загрузки DDS есть параметр forceSRGB, но вы туда передает false
5) Настройте, пожалуйста, камеру так, чтобы она по умолчанию показывала что-нибудь осмысленное. А то сходу и непонятно, почему ничего нет на картинке. 

В общем, получается, несколько мелких огрехов, которые в итоге ведут к тому, что нужный алгоритм не работает. Хотя наибольшая по объему часть работы вполне себе выполнена. Если исправите сегодня, засчитаю задачу, как выполненную вовремя.

anthark reviewed (changes requested):
Написал один большой комментарий, надеюсь, вы его увидите.

maik-nack committed:
Made some changes

- Changed camera position and view direction
- Changed format of render texture
- Changed texture reading to sRGB
- Changed eye adaptation
- Added CPU texture unmapping

leins275 commented:
Мы пробовали в формуле адаптации использовать среднюю яркость и обычную, и логарифмическую, разница не заметна.

anthark commented:
> 
> 
> Мы пробовали в формуле адаптации использовать среднюю яркость и обычную, и логарифмическую, разница не заметна.

Так это в вашем случае - один источник света, определенной яркости. А если их будет больше и с большей интенсивностью? Возможно, адаптация станет слишком быстрой или слишком медленной. Давайте все-таки делать так, как нам предлагают авторы алгоритма =)

maik-nack committed:
Changed average luminance calculating

maik-nack committed:
Changed luminance to logarithm
